src/compiler/scala/tools/nsc/backend/jvm/opt/Inliner.scala

```diff
@@ -142,6 +142,7 @@ class Inliner[BT <: BTypes](val btypes: BT) {
           elided += r
         else
           result += r
+        ()
```

Comment:
The improvements to LUBs now compute a sharper LUB for this expression, but that fails in bound checks.